R/internals.R

```diff
@@ -379,3 +379,20 @@
 #   if(getOption("mypackage.verbose", default = TRUE)) message(glue::glue(msg))
 #   return(NULL)
 # }
+
+
+#' Load Page with headers
+#'
+#' loads webpages with a header passed to read_html
+#'
+#' @param page_url url of the page wanted to be loaded
+#'
+#' @return a html webpage
+#'
+#' @importFrom magrittr %>%
+#' @noRd
+#'
+.load_page <- function(page_url) {
+  ua <- httr::user_agent("RStudio Desktop (2022.7.1.554); R (4.1.1 x86_64-w64-mingw32 x86_64 mingw32)")
```

Comment:
i would do something like this

```r
version_info <- rstudioapi::versionInfo()
session_info <- sessionInfo()

client <- if(interactive() & rstudioapi::isAvailable()) {
  sprintf(
    "RStudio %s (%s); %s %s",
    version_info$mode,
    version_info$version,
    session_info$R.version$version.string,
    session_info$R.version$platform
  )
} else {
  sprintf(
    "%s %s",
    session_info$R.version$version.string,
    session_info$R.version$platform
  )
}

ua <- httr::user_agent(client)
```

of course, this does introduce a new dependency (`{rstudioapi}`), so there is a little downside. additionally, we would be running this every time a user wants to load a new page. to avoid repeated `rstudioapi` calls, i would implement some globals to track the `client` in a session, similar to what's implemented [here](https://github.com/collegevine/lightMLFlow/blob/main/R/globals.R)

Reply:
I like this solution, but it makes things tricky for jobs running on GitHub Actions...

Trying to run this MRE as a github action:

```
version_info <- rstudioapi::versionInfo()
session_info <- utils::sessionInfo()

client <- if(interactive() & rstudioapi::isAvailable()) {
  sprintf(
    "RStudio %s (%s); %s %s",
    version_info$mode,
    version_info$version,
    session_info$R.version$version.string,
    session_info$R.version$platform
  )
} else {
  sprintf(
    "%s %s",
    session_info$R.version$version.string,
    session_info$R.version$platform
  )
}

ua <- httr::user_agent(client)

print(client)

stat_urls <- "https://fbref.com/en/comps/Big5/2021-2022/stats/players/2021-2022-Big-5-European-Leagues-Stats"
team_page <- xml2::read_html(rvest::session(stat_urls, ua))

head(team_page)
```

With this yml:

```
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.
#
# See https://github.com/r-lib/actions/tree/master/examples#readme for
# additional example workflows available for the R community.

name: R

on:
  push:
    branches: [ "master" ]
  pull_request:
    branches: [ "master" ]

permissions:
  contents: read

jobs:
  build:
    runs-on: macOS-latest
    # retrieve token
    env:
      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
      # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2
      - uses: r-lib/actions/setup-r@master
      - name: Package Installation
        run: Rscript -e 'install.packages(c("rstudioapi", "utils", "httr", "xml2", "rvest"))'
      - name: Run set agent
        run: Rscript -e 'source("test/test2.R", echo = TRUE)'
```
Leads to this error:

<img width="588" alt="image" src="https://user-images.githubusercontent.com/31471783/185723539-954c4494-ae47-4793-9c45-41274437f34e.png">

Reply:
I see. I can verify that. I tried a slightly modified version of your test, where we default to `"RStudio Desktop (2022.7.1.554)"` for the first part of the string, but use the user's platform information to compose the second part of the string (with `utils::sessionsInfo()`, as before. See [this successful GHA run](https://github.com/JaseZiv/worldfootballR/runs/7931824815?check_suite_focus=true).

Reply:
I've put my suggestion for using a package option to save the `agent` for all calls to `.load_page()` in an alternate PR: #171 . Feel free to not use that PR, or maybe to make the same edits in this PR.

Reply:
Ok cool. I’ll have a look at this when I’m back in front of my screen

Reply:
I have incorporated your suggestions in this PR.

Additionally, I reverted page loads for `fb_player_scouting_report()`  back to `xml2::read_html()`, rather than `.page_load()` for one reason - I think some users absolutely abuse this function and try to get scouting reports for whole leagues at a time (or the whole of the Big 5) and I want to minimise the risk of these headers also ending up on a blocked list... thoughts @tonyelhabr?